staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -235,41 +235,64 @@ func (c *AvailableConditionController) sync(key string) error {
 	}
 	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
 	if apiService.Spec.Service != nil && c.serviceResolver != nil {
-		discoveryURL, err := c.serviceResolver.ResolveEndpoint(apiService.Spec.Service.Namespace, apiService.Spec.Service.Name)
-		if err != nil {
-			return err
+		attempts := 5
+		results := make(chan error, attempts)
+		for i := 0; i < attempts; i++ {
+			go func() {
+				discoveryURL, err := c.serviceResolver.ResolveEndpoint(apiService.Spec.Service.Namespace, apiService.Spec.Service.Name)
```

Comment:
if this resolves to the same endpoint every time, do we still want to attempt to fetch it 5 times?

Reply:
should this API change to return a slice of resolved endpoints, so we can try each one instead of depending on rand() to hit all of them?

Reply:
> if this resolves to the same endpoint every time, do we still want to attempt to fetch it 5 times?

Yes.  The endpoint is allowed to be a DNS name or a service IP, which would roundrobin.

Reply:
unconditionally hitting whatever endpoint comes back five times just in case it resolves differently seems weird

Reply:
I guess there are single-backend scenarios where this does make sense (e.g. a single endpoint that resolves to a single backend hit a flake and this would avoid making it unavailable for 30 seconds if 1 of N requests made it through)